### PR TITLE
Use the node v8 promisify API to remove unnecessary code

### LINKED ISSUE #153: Use the node v8 promisify API to remove unnecessary code
We can remove some unnecessary explicit `promisify` code by using the v8 `utils.promisify` API (https://nodejs.org/en/blog/release/v8.0.0/#improved-support-for-promises). 

E.g.: This [function](https://github.com/zalando/tailor/blob/master/lib/fetch-template.js#L29) can be changed to a much elegant and declarative implementation.

```javascript
const readFile = (path) => 
    new Promise((resolve, reject) => {
        fs.readFile(path, 'utf-8', (err, data) => {
            if (err) {
                reject(new TemplateError(err));
                return;
            } 
            resolve(data);
        });
    });
```
to

```javascript
const util = require('util');

const readFile = (path) => 
    util.promisify(fs.readFile)(path)
        .catch(err => new TemplateError(err));
```

### PR DESCRIPTION
**Pull request by addityasingh**
- closes #153, #172 
- Use the node 8 promisify API to remove unnecessary code
- Polyfill it for older node versions

### CODE REVIEW

**Comment by vigneshshanmugam** [comment]
@addityasingh This one looks good. can you also add tests for fetch-template? You had a PR with tests and removed it at some point. remember?

**Comment by addityasingh** [comment]
I created a separate PR for it earlier and it was already merged. See https://github.com/zalando/tailor/pull/161

**Comment by vigneshshanmugam** [comment]
👍

**Comment by addityasingh** [comment]
👍

File: package.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
   "license": "MIT",
   "dependencies": {
     "http-link-header": "^0.8.0",
-    "parse5": "^3.0.2"
+    "parse5": "^3.0.2",
+    "util.promisify": "^1.0.0"
```
Comment (vigneshshanmugam) [inline package.json]: oops what about yarn.lock? can you generate it?

Comment (addityasingh) [reply package.json]: Done

**Comment by addityasingh** [comment]
:

**Comment by vigneshshanmugam** [comment]
👍

**Pull request merged**